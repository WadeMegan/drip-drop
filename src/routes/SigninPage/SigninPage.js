import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './SigninPage.css'
import SigninForm from '../../components/SigninForm/SigninForm'
import Error from '../../components/Error/Error'
import PlantListContext from '../../contexts/PlantListContext'

export default class SigninPage extends Component {
    static contextType = PlantListContext

    componentWillUnmount(){
        this.context.clearError()
    }

    // has been passed onLogin props from App component and is passing to SigninForm component
    render() {
        return (
            <Error>
            <section className = 'signinSection'>
                <SigninForm onLogin={this.props.onLogin}/>
            </section>
            </Error>
        )
    }
}

