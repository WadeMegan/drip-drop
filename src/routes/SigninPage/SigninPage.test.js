import React from 'react'
import ReactDOM from 'react-dom'
import SigninPage from './SigninPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SigninPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SigninPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
