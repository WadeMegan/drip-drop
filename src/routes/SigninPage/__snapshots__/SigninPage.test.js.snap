// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<section
  className="signinSection"
>
  <form
    className="signinForm"
    onSubmit={[Function]}
  >
    <h2
      className="formTitle"
    >
      Welcome Back
    </h2>
    <div
      className="formElements"
    >
      <p
        className="demo"
      >
        To demo, sign in with email: test@gmail.com and password: password
      </p>
      <div
        id="error"
        role="alert"
      />
      <div>
        <label
          htmlFor="username"
        >
          Email
        </label>
        <input
          id="username"
          name="username"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
        />
      </div>
      <input
        className="formSigninButton"
        type="submit"
        value="Sign In"
      />
      <p
        className="createAccountLink"
      >
        New? 
        <a
          href="/register"
          onClick={[Function]}
        >
          Create account.
        </a>
      </p>
    </div>
  </form>
</section>
`;
